src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -2787,7 +2787,7 @@ void AdapterHandlerLibrary::create_native_wrapper(const methodHandle& method) {
           }
         }
 
-        DirectiveSet* directive = DirectivesStack::getDefaultDirective(CompileBroker::compiler(CompLevel_simple));
+        DirectiveSet* directive = DirectivesStack::getMatchingDirective(method, CompileBroker::compiler(CompLevel_simple));
```

Comment:
Will it work for when `-XX:+PrintAssembly` used and not `-XX:CompileCommand=print,`?

Reply:
Yes, it does, because `-XX:+PrintAssembly` sets `PrintAssembly` to true in the default compiler directives so `DirectivesStack::getMatchingDirective()` will return true for every method. There's also a test for this case in `NativeCalls.java`:
```cpp
 new Variant(List.of("-XX:-TieredCompilation",
                     "-XX:+UnlockDiagnosticVMOptions", "-XX:+PrintAssembly"), "true", "true"),
```
I've now also added another test for `-XX:+PrintNativeNMethods` which as you've correctly observed, also prints native methods, but all of them:
```cpp
 new Variant(List.of("-XX:-TieredCompilation",
                     "-XX:+UnlockDiagnosticVMOptions", "-XX:+PrintNativeNMethods"), "true", "true"),
```